pkg/secrets/secrets.go

```diff
@@ -102,6 +102,16 @@ type SecretsDriver interface {
 	Delete(id string) error
 }
 
+// StoreOptions are optional metadata fields that can be set when storing a new secret
+type StoreOptions struct {
+	// DriverOptions is other metadata needed to use the driver
+	DriverOpts map[string]string
+	// Metadata stores other metadata on the secret
+	Metadata map[string]string
```

Comment:
```suggestion
	// DriverOptions are extra options used to run this driver
	DriverOpts map[string]string
	// Metadata stores extra metadata on the secret
	Metadata map[string]string
```
Maybe something like this?